### PR TITLE
Ensure that _prom_ext schema created safely

### PR DESCRIPTION
**Pull request by JamesGuthrie**
Because the extension was declared as non-relocatable, and in the schema
`_prom_ext`, it was possible to have that schema (and objects in it)
created before the extension install script is run. This opened the
extension up to precreation attacks.

This change takes a number of measures to mitigate that.

- We configure the schema to `public`, which allows the extension to
  control the creation of the `_prom_ext` schema.
- We add a `bootstrap` migration script which is run before the pgx
  generated SQL, which creates the `_prom_ext` schema.
- We tell pgx to put all generated objects in the `_prom_ext` schema.

### CODE REVIEW

**Review by sumerman — APPROVED** [review]
Almost LGTM with a couple of caveats:
- I only visually inspected SQL changes trying to assess whether changes fall into one of the few common patterns. But I didn't read through every single change.
- I assume that Rust code changes outside of `src/schema` are limited to adding `#[pg_schema] mod _prom_ext`, fixing tests and reformatting.

File: build.rs [hunk 1]
```diff
@@ -33,17 +33,32 @@ fn main() {
...
+    let mut out_path = PathBuf::from(manifest_dir.clone());
     out_path.push(MIGRATION_FILE_NAME);
     let mut file = File::create(out_path).unwrap();
-    file.write_all(sql.as_bytes())
+    file.write_all(rendered_sql.as_bytes())
         .expect("unable to write file contents");
+
+    let mut bootstrap_path = PathBuf::from(manifest_dir);
+    bootstrap_path.push("bootstrap.sql");
+    let mut migration_sql = include_str!("migration/migration-table.sql").to_string();
```
Comment (cevian) [inline build.rs:50]: should we move these files into
- migration/bootstrap/migration-table.sql
- migration/bootstrap/schema.sql
?

**Review by cevian — APPROVED** [review]
This review is only of the last commit. LGTM, some nits below

**Review by cevian — APPROVED** [review]

File: sql/promscale--0.0.0.sql [hunk 2]
```diff
@@ -435,6 +435,7 @@ BEGIN
     -- Bring migrations table up to speed
     INSERT INTO _ps_catalog.migration (name, applied_at_version)
     VALUES
+        ('create_ext_schema.sql'          , '0.0.0'),
```
Comment (jgpruitt) [inline sql/promscale--0.0.0.sql:438]: the fact that this isn't 000_ makes my eye twitch, but I'll get over it :)

File: build.rs [hunk 3]
```diff
@@ -76,15 +78,33 @@ fn generate_control_file(manifest_dir: &str) {
...
     out_path.push(MIGRATION_FILE_NAME);
     let mut file = File::create(out_path).unwrap();
-    file.write_all(sql.as_bytes())
-        .expect("unable to write migration file contents");
+    file.write_all(rendered_sql.as_bytes())
+        .expect("unable to write file contents");
+
+    let mut bootstrap_out_path = PathBuf::from(manifest_dir);
+    bootstrap_out_path.push(BOOTSTRAP_FILE_NAME);
+    let mut bootstrap_out_file = File::create(bootstrap_out_path).unwrap();
```
Comment (jgpruitt) [inline build.rs:91]: Add this path to .gitignore, please?

**Review by jgpruitt — CHANGES_REQUESTED** [review]
only 1 nit

File: build.rs (hunk 3 above)
Comment (cevian) [reply build.rs:91]: done

**Review by jgpruitt — APPROVED** [review]

File: migration/bootstrap.sql [hunk 4]
```diff
@@ -0,0 +1 @@
+CREATE SCHEMA _prom_ext;
```
[the remark this answers is not in the document]
Comment (JamesGuthrie) [reply migration/bootstrap.sql]: done

File: build.rs (hunk 1 above)
Comment (JamesGuthrie) [reply build.rs]: done

**Pull request merged**